src/js/controller/controller.js

```diff
@@ -336,15 +340,30 @@ Object.assign(Controller.prototype, {
         function _checkPlayOnViewable(model, viewable) {
             if (model.get('playOnViewable')) {
                 if (viewable) {
-                    _autoStart();
-                } else if (OS.mobile || (model.get('autoPause') && model.get('autoPause').viewability)) {
+                    _play('viewable');
```

Comment:
The problem with this is that autostart players still depend on "playOnViewable". They really shouldn't. If they are supposed to call `_autoStart('viewable');` then a separate callback, or some additional logic here would allow us to call either `_autoStart('viewable');` or `_play('viewable');` as appropriate.

In the model this is happening (not ideal since `_autoStart` should only be called once):
```js
    // Mobile players always wait to become viewable.
    // Desktop players must have autostart set to viewable
    setAutoStart(autoStart) {
        if (autoStart !== undefined) {
            this.set('autostart', autoStart);
        }

        const autoStartOnMobile = OS.mobile && this.get('autostart');
        this.set('playOnViewable', autoStartOnMobile || this.get('autostart') === 'viewable');
```